Jamfile

```diff
@@ -376,6 +377,12 @@ rule building ( properties * )
 		}
 	}
 
+	if [ configure.builds /boost/architecture//x86 : $(properties) : x86 ]
+		&& ! <target-os>windows in $(properties)
```

Comment:
I think it's probably better to only include toolsets known to support it.. I bet metrowerks, xlc, sunstudio don't support the option. Also, specifying this flag, won't that make GCC/clang use popcnt by itself as well, basically assuming the hardware supports it. It's important that _only_ the explicit calls to it generate this instruction, which is protected by the run-time check of support for it.